Masonry

gbjbaanb opened the pull request:
added Masonry js library for framedex-style layout

gbjbaanb committed:
Hide category headings

When all items have been hidden, hide the categories too

gbjbaanb committed:
added Masonry link

gbjbaanb committed:
masonry layout changes

ehwuts commented:
I'm very conflicted about this one, at least without a maintaining access to both layouts, especially since I personally prefer the existing layout.

As is it's not exactly living up to Masonry's claims
> Masonry is a JavaScript grid layout library. It works by placing elements in optimal position based on available vertical space, sort of like a mason fitting stones in a wall. You’ve probably seen it in use all over the Internet.

Especially on mobile, where the floating columns layout (at least currently) seems **very** confused about how do deal with the large disparities in section length. Although I'm entirely unfamiliar with Masonry, so I don't know how much of this is caused by the existing page formatting being an evening of handrolled jank.

<img src="https://user-images.githubusercontent.com/4962928/126235438-bdb0890d-2208-462d-80d4-c8a3e75d0cce.png" width="150" height="300">

ehwuts commented:
Changing the config button text color from `eee` to `808080` also seems a bit too low contrast against the `103` background, though that's relatively minor.

gbjbaanb commented:
I just changed it around, I was disappointed with Masonry - but I think it needs a height specifying for each block. Still, fite more onto a single page once you have a fair number hidden, like Framedex, which was my goal.

gbjbaanb committed:
Add files via upload

changed to use CSS only using the multicol spec. Seems to work OK, and fully responsive.

gbjbaanb committed:
Add files via upload

quick bufix: reset entries needs to toggle any hidden categories

gbjbaanb commented:
I've updated this with a CSS only solution that works somewhat better than Masonry. The css file has media queries, but I think they could go and only keep the last spec for .listings that creates a number of columns automatically to fit the screen (possibly make the column-width property wider)

Still, your choice if this framedex-style-view is a better display than the original, but its just css now, so could even be togglable.

ehwuts commented:
The tidiness of alignments in new version removes most of my objection to switching designs, though my personal preference still wants to retain the option to use the old layout. 

As for adding such a toggle.. it's been long enough that I don't remember if giving a `link` stylesheet an id for easy access and just hotswapping the `href` value triggers a well-behaved page rendering update, or if that would misbehave and it would require a more involved implementation.

Aside from that, new layout seems to need a bit more tweaking for consistent cross-browser rendering. Right now, on a 1900x1080 desktop display, I'm getting a tidy 6 columns filling my screen on firefox, while in chromium and edge I only get 3 columns though they're spaced as if to make room for 3 more. 
![image](https://user-images.githubusercontent.com/4962928/126723463-68bfd6c7-7771-44ac-9f31-4a689bcbf907.png)
![image](https://user-images.githubusercontent.com/4962928/126723431-a34ac067-ad1e-4740-87db-10c9a1200d69.png)
![image](https://user-images.githubusercontent.com/4962928/126724298-56d65cd8-5cfa-481c-81ec-38a3e2f4fcaf.png)

Meanwhile android chrome on my phone also has 3 columns in both landscape and portrait.. but since my phone seems to be stuck treating landscape as zoomed in portrait and the spacing works out that's probably still working reasonably well in both cases.
<img src="https://user-images.githubusercontent.com/4962928/126724569-bf454a0d-561d-4857-970a-f8ee7f457e67.png" width="150" height="300"><img src="https://user-images.githubusercontent.com/4962928/126724572-8e445cf1-8d97-490a-ae12-6fdcd34e1485.png" width="300" height="150">

ehwuts commented:
..or I'm overthinking things and style swapping could be handled purely by toggling an addition class's presence on the wrapper to switch rule precedence

gbjbaanb commented:
Its only a quick modificiation, and I use Firefox so never really tried in Chrome (which also refuses to load the json with a CORS error so I didn't test it out previously). 

Change the .category css to use `display: block `rather than `inline-block `and it works as intended on Chrome (and FF). No idea why that is such a difference, and inline-block was suggetsed as the thing to use on the internet doc I read, but that's modern web for you.

(further testing shows that display:block works great for Chrome, but wraps columns in FF. display:inline-block works great for FF, but as youy see, reduces it to 3 columns. What a pain)

(TBH I think its all a bit flaky, I'm going to keep it on a local copy as it works for me, but then I've ticked off most of the things anyway. Might be worth keeping the .category and listings changes as a comment - maybe one day we'll be able to have nice things :) )

To return to row-based, just remove the CSS .listings and .category changes, these were empty in your UI, all I did was put styles in them.
